trotters/trotters.py

```diff
@@ -0,0 +1,259 @@
+        """
+        Lists all roles which can be added/removed
+        """
+        enabled = list()
+        for role in ctx.guild.roles:
+            if await self.config.role(role).addable() and not role.is_default():
+                enabled.append(role.name)
+        await ctx.maybe_send_embed(f"List of addable roles: \n%s" % (", ".join(enabled)))
+
+    @commands.bot_has_permissions(manage_roles=True)
+    @trotters.command(name="add")
+    async def add_role(self, ctx: GuildContext, user_id: int, role: discord.Role, *, reason: str = "Added by command."):
+        """
+        Adds specified role to given user
+        """
+
+        if not await self.config.role(role).addable():
+            return await ctx.maybe_send_embed("Role isn't set as addable.")
+        try:
+            member = ctx.guild.get_member(user_id)
+            if role in member.roles:
+                return await ctx.maybe_send_embed("Member already has that role.")
+
+            data = await self.config.role(role).USERS()
+            if len(ctx.message.attachments):
+                attachment = ctx.message.attachments[0]
+                case = await modlog.create_case(
+                    self.bot,
+                    member.guild,
+                    ctx.message.created_at,
+                    "Role List Update",
+                    member,
+                    moderator=ctx.author,
+                    reason=(f"{reason}. {attachment.url}"),
+                )
+            else:
+                pred = MessagePredicate.yes_or_no(ctx)
+                await ctx.send(f"Couldn't find attachment, do you want to continue without adding attachment?")
+                try:
+                    await self.bot.wait_for("message", check=pred, timeout=30)
+                except asyncio.TimeoutError:
+                    return await ctx.send("Timed out.")
+                if pred.result:
+                    case = await modlog.create_case(
+                        self.bot,
+                        member.guild,
+                        ctx.message.created_at,
+                        "Role List Update",
+                        member,
+                        moderator=ctx.author,
+                        reason=(f"{reason}. Missing attachment."),
+                    )
+                else:
+                    return await ctx.maybe_send_embed("Cancelling command.")
+            caseno = case.case_number
+            await member.add_roles(role)
+            data[member] = caseno
+            await self.config.role(role).USERS.set(data)
+            await ctx.tick()
+        except discord.Forbidden:
+            return await ctx.maybe_send_embed("Can't do that. Discord role heirarchy applies here.")
+
+    @commands.bot_has_permissions(manage_roles=True)
+    @trotters.command(name="remove")
+    async def remove_role(
+        self, ctx: GuildContext, user_id: int, role: discord.Role, *, reason: str = "Removed by command."
```

Comment:
again, change the user_id to discord.Member object. also change the reason to include cog name by default.